scripts/copy_from_upstream/requirements.txt

```diff
@@ -1,12 +1,153 @@
-attrs==20.3.0
-GitPython==3.1.41
-importlib-metadata==3.7.0
-Jinja2==3.1.3
-markdown-it-py==2.2.0
-MarkupSafe==2.1.3
-mdit-py-plugins==0.3.4
-PyYAML==6.0.1
-tabulate==0.8.10
-typing-extensions==3.7.4.3
-wget==3.2
-zipp==3.4.0
+attrs==20.3.0 \
+    --hash=sha256:31b2eced602aa8423c2aea9c76a724617ed67cf9513173fd3a4f03e3a929c7e6 \
```

Comment:
Please add documentation how to generate these hashes (and how to handle any errors they may trigger).

Reply:
Agreed - I'll answer in more detail below

Reply:
Now added in PROCEDURES.md

Reply:
Thanks for the documentation! Our etiquette so far was that comments are resolved by the people that raised them. Please let me know if that doesn't suit you.